internal/shared/kubernetes/kubernetes.go

```diff
@@ -75,3 +83,33 @@ func GetSecret(ctx context.Context, client k8sClient.Client, namespace, secretNa
 	}
 	return secret, nil
 }
+
+// ApplyOwnershipReference adds  or updates ownerships to kubernetes resources in GitSync CRD
```

Comment:
```suggestion
// ApplyOwnershipReference sets the GitSync as the Owner of the Resource through an OwnerReference
```

Reply:
This also actually appends to existing owner if already exists, right? may want to make sure the comment also covers that.

`// ApplyOwnershipReference sets the GitSync as the Owner (or one of the Owners if ones already exist) of the Resource through an OwnerReference`